components/sbm-core/src/test/java/org/springframework/sbm/build/migration/actions/RemoveManagedDependenciesTest.java

```diff
@@ -0,0 +1,71 @@
+package org.springframework.sbm.build.migration.actions;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.sbm.build.api.Dependency;
+import org.springframework.sbm.engine.context.ProjectContext;
+import org.springframework.sbm.project.resource.TestProjectContext;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class RemoveManagedDependenciesTest {
+
+    @Test
+    public void givenProjectWithManagedDependency_removeSpringManagedDependencies_expectHibernateDependencyRemoved(){
+        final String hibernateCoordinates = "org.hibernate:hibernate-core:5.6.11.Final";
+        final String springBootDataJpaCoordinates = "org.springframework.boot:spring-boot-starter-data-jpa:2.7.4";
+
+        final ProjectContext projectContext = TestProjectContext.buildProjectContext()
+                .withBuildFileHavingDependencies(hibernateCoordinates, springBootDataJpaCoordinates)
+                .build();
+
+        RemoveManagedDependencies removeManagedDependencies = new RemoveManagedDependencies();
+        removeManagedDependencies.apply(projectContext);
+
+        assertThat(projectContext.getBuildFile()
+                .getDeclaredDependencies()
+                .stream()
+                .map(Dependency::getCoordinates)
+                .anyMatch(hibernateCoordinates::equals)
+        ).isFalse();
+    }
+
+    @Test
+    public void givenProjectWithLowerVersionedManagedDependency_removeSpringManagedDependencies_expectDependencyRemoved(){
+        final String hibernateCoordinates = "org.hibernate:hibernate-core:5.3.2.Final";
+        final String springBootDataJpaCoordinates = "org.springframework.boot:spring-boot-starter-data-jpa:2.7.4";
+
+        final ProjectContext projectContext = TestProjectContext.buildProjectContext()
+                .withBuildFileHavingDependencies(hibernateCoordinates, springBootDataJpaCoordinates)
+                .build();
+
+        RemoveManagedDependencies removeManagedDependencies = new RemoveManagedDependencies();
+        removeManagedDependencies.apply(projectContext);
+
+        assertThat(projectContext.getBuildFile()
+                .getDeclaredDependencies()
+                .stream()
+                .map(Dependency::getCoordinates)
+                .anyMatch(hibernateCoordinates::equals)
+        ).isFalse();
```

Comment:
This results to true in the CI

Reply:
The problem was with the condition Dependency.isRecentThen(that). I have fixed the code and test is passing now.